components/latex_block/latex_block.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import katex from 'katex';
+
+type Props = {
+    content: string;
+    enableLatex: boolean;
+};
+
+type KatexOptions = {
+    throwOnError?: boolean;
+    displayMode?: boolean;
+    maxSize?: number;
+    maxExpand?: number;
+    fleqn?: boolean;
+};
+
+type Output = {
+    latex: string;
+};
+
+function LatexBlock({
+    content,
+    enableLatex = false,
+}: Props): JSX.Element {
+    const [katexOptions, setKatexOptions] = useState<KatexOptions>();
+
+    const [output, setOutput] = useState<Output>({
+        latex: '',
+    });
+
+    const [hasError, setHasError] = useState<boolean>(false);
+
+    useEffect(() => {
+        try {
+            setKatexOptions({
+                throwOnError: false,
+                displayMode: true,
+                maxSize: 200,
+                maxExpand: 100,
+                fleqn: true,
+            });
+
+            setOutput({
+                latex: katex.renderToString(content, katexOptions),
+            });
+        } catch (e) {
+            setHasError(true);
+        }
+    }, []);
+
+    return (
```

Comment:
There are a few small issues with the return here. 

If you had an error when running `katex.renderToString(content, katexOptions)`, this would still try to render:
```
<div
      dangerouslySetInnerHTML={{
            __html: output.latex,
      }}
/>
```

Also, previously I think our latex would have rendered like this:
```
<div
     className='post-body--code tex'
>
     dangerouslySetInnerHTML={{__html: html}}
</div>
```
but now it will render like so:
```
<div
     className='post-body--code tex'
>
     <div>
            dangerouslySetInnerHTML={{__html: html}}
     </div>
</div>
```